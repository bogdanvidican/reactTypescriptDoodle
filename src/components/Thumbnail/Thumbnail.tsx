import * as React from 'react';
import { WithStyles, withStyles, GridListTile, GridListTileBar, createStyles } from '@material-ui/core';

type ClassKeys = 'tile';

export interface IProps {
  title: string,
  thumb: string,
}

export interface InnerProps extends IProps, WithStyles<ClassKeys> {}

const style = () => createStyles({
  tile: {
    width: 200,
    height: 200,
    margin: 12.5,
    borderRadius: '4px',
  },
});

export const ThumbnailComponent = ({ title, thumb, classes }: InnerProps) => {
  return (
    <GridListTile classes={{ tile: classes.tile }}>
      <img src={thumb} alt={title} />
      <GridListTileBar
        title={title}
      />
    </GridListTile>
  )
};

export const Thumbnail = withStyles(style)(ThumbnailComponent);
