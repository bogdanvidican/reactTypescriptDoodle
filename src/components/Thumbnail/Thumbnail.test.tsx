import * as React from 'react';
import { shallow } from 'enzyme';
import { ThumbnailComponent } from './Thumbnail';
import { InnerProps } from './Thumbnail';
import { GridListTileBar } from '@material-ui/core';

const props: InnerProps = {
    thumb: 'https://some.test.thumb',
    title: 'some test title',
    classes: {
      tile: 'tile'
    }
}

let wrapper: any;

beforeEach(() =>{
  wrapper = shallow(<ThumbnailComponent {...props} />);
})
  
describe('<Thumbnail />', () => {
  const { thumb, title } = props;
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('uses correct thumb src', () => {
    expect(wrapper.find('img').prop('src'))
      .toEqual(thumb);
  });
  it('has correct title displayed', () => {
    expect(wrapper.find(GridListTileBar).prop('title'))
      .toEqual(title);
  });
});
