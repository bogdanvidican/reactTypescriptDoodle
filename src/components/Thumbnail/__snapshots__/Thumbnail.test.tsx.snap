// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail /> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "tile": "tile",
          }
    }
    thumb="https://some.test.thumb"
    title="some test title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="some test title"
          src="https://some.test.thumb"
/>,
        <WithStyles(GridListTileBar)
          title="some test title"
/>,
      ],
      "classes": Object {
        "tile": "tile",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "some test title",
          "src": "https://some.test.thumb",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "some test title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="some test title"
            src="https://some.test.thumb"
/>,
          <WithStyles(GridListTileBar)
            title="some test title"
/>,
        ],
        "classes": Object {
          "tile": "tile",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "some test title",
            "src": "https://some.test.thumb",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "some test title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
